build/signedbuild.yml

```diff
@@ -17,347 +17,376 @@ variables:
-    displayName: 'TSA upload (BinSkim)'
-    condition: and(succeeded(), eq(variables.isMain, true))
-    inputs:
-      GdnPublishTsaOnboard: true
-      GdnPublishTsaConfigFile: '$(Build.SourcesDirectory)\tsa.config'
-
-  - task: CopyFiles@2
-    displayName: 'Copy Files to: $(Build.ArtifactStagingDirectory)'
-    inputs:
-      Contents: '**\bin\release\**'
-      TargetFolder: '$(Build.ArtifactStagingDirectory)'
-
-  - task: CopyFiles@2
-    displayName: 'Copy MSI File to: $(Build.ArtifactStagingDirectory)'
-    inputs:
-      Contents: '**\CLI_Installer\bin\x86\Release\**'
-      TargetFolder: '$(Build.ArtifactStagingDirectory)'
-
-  - task: PublishBuildArtifacts@1
-    displayName: 'Publish Artifact: drop'
-
-  - task: PublishBuildArtifacts@1
-    displayName: 'Publish Artifact: NuGet package'
-    inputs:
-      PathtoPublish: 'src\CI\bin\Release\NuGet'
-      ArtifactName: 'NuGet'
-
-  - task: PublishBuildArtifacts@1
-    displayName: 'Publish Artifact: Rules markdown'
-    inputs:
-      PathtoPublish: 'src\RulesMD\bin\Release\axe-windows-rules.md'
-      ArtifactName: 'axe-windows-rules'
-
-  - task: PublishBuildArtifacts@1
-    displayName: 'Publish Artifact: CLI (msi)'
-    inputs:
-      PathtoPublish: 'src\CLI_Installer\bin\x86\Release\AxeWindowsCLI.msi'
-      ArtifactName: 'CLI-msi'
-
-  - task: PublishBuildArtifacts@1
-    displayName: 'Publish Artifact: CLI (zip)'
-    inputs:
-      PathtoPublish: 'src\CLI_Installer\bin\x86\Release\AxeWindowsCLI.zip'
-      ArtifactName: 'CLI-zip'
-
-  - task: CopyFiles@2
-    displayName: 'Copy CLI_Full Files for Signing Validation'
-    inputs:
-      SourceFolder: '$(Build.SourcesDirectory)\src'
-      Contents: |
-       CLI_Full\bin\release\net6.0\win7-x86\?(*.exe|*.dll)
-      TargetFolder: '$(Build.ArtifactStagingDirectory)\SigningValidation'
-
-  - task: CopyFiles@2
-    displayName: 'Copy CLI Files for Signing Validation'
-    inputs:
-      SourceFolder: '$(Build.SourcesDirectory)\src'
-      Contents: 'CLI\bin\release\net6.0\**\?(*.exe|*.dll)'
-      TargetFolder: '$(Build.ArtifactStagingDirectory)\SigningValidation'
-
-  - task: ms-vseng.MicroBuildTasks.521a94ea-9e68-468a-8167-6dcf361ea776.MicroBuildCleanup@1
-    displayName: 'Perform Cleanup Tasks'
-    condition: succeededOrFailed()
+resources:
+  repositories:
+  - repository: MicroBuildTemplate
+    type: git
+    name: 1ESPipelineTemplates/MicroBuildTemplate
+    ref: refs/tags/release
+
+extends:
+  template: azure-pipelines/MicroBuild.1ES.Official.yml@MicroBuildTemplate
+  parameters:
+    pool: 
+      name: MSEngSS-MicroBuild2022-1ES
+    sdl:
+      binskim:
+        analyzeTargetGlob: "$(System.DefaultWorkingDirectory)\\src\\CI\\bin\\**\\*.dll;\
+                            $(System.DefaultWorkingDirectory)\\src\\Actions\\bin\\**\\*.dll;\
+                            $(System.DefaultWorkingDirectory)\\src\\Automation\\bin\\**\\*.dll;\
```

Comment:
I understand why we're duplicating the glob list, but it opens the door to the two lists getting out of sync. I'd love to see this combined into a single statement, but a reasonable alternative would be to add a comment before each `analyzeTargetGlob` block to be sure that all changes gets reflected in both places. Maybe something like "If you modify this list, you also need to modify the list in the Run BinSkim task"

Reply:
Added the comment